src/dynarec/rv64/dynarec_rv64_avx_0f.c

```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <errno.h>
+
+#include "debug.h"
+#include "box64context.h"
+#include "dynarec.h"
+#include "emu/x64emu_private.h"
+#include "emu/x64run_private.h"
+#include "x64run.h"
+#include "x64emu.h"
+#include "box64stack.h"
+#include "callback.h"
+#include "emu/x64run_private.h"
+#include "x64trace.h"
+#include "dynarec_native.h"
+#include "my_cpuid.h"
+#include "emu/x87emu_private.h"
+#include "emu/x64shaext.h"
+
+#include "rv64_printer.h"
+#include "dynarec_rv64_private.h"
+#include "dynarec_rv64_functions.h"
+#include "dynarec_rv64_helper.h"
+
+uintptr_t dynarec64_AVX_0F(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ninst, vex_t vex, int* ok, int* need_epilog)
+{
+    (void)ip; (void)need_epilog;
+
+    uint8_t opcode = F8;
+    uint8_t nextop, u8;
+    uint8_t gd, ed;
+    uint8_t wback, wb1, wb2, gback;
+    uint8_t eb1, eb2, gb1, gb2;
+    int32_t i32, i32_;
+    int cacheupd = 0;
+    int v0, v1, v2;
+    int q0, q1, q2;
+    int d0, d1, d2;
+    int s0;
+    uint64_t tmp64u;
+    int64_t j64;
+    int64_t fixedaddress, gdoffset;
+    int unscaled;
+    MAYUSE(wb1);
+    MAYUSE(wb2);
+    MAYUSE(eb1);
+    MAYUSE(eb2);
+    MAYUSE(gb1);
+    MAYUSE(gb2);
+    MAYUSE(q0);
+    MAYUSE(q1);
+    MAYUSE(d0);
+    MAYUSE(d1);
+    MAYUSE(s0);
+    MAYUSE(j64);
+    MAYUSE(cacheupd);
+    #if STEP > 1
+    static const int8_t mask_shift8[] = { -7, -6, -5, -4, -3, -2, -1, 0 };
+    #endif
+
+    rex_t rex = vex.rex;
+
+    switch(opcode) {
+
+        case 0x10:
+            INST_NAME("VMOVUPS Gx,Ex");
+            nextop = F8;
+            GETGX();
+            GETEX(x2, 0, 8);
+            SSE_LOOP_MV_Q(x3);
+            if(vex.l) {
+                GETGY();
+                GETEY(x2, 0, 16);
```

Comment:
Hmm, this seems not correct, how is the 16 calculated? Also, `GETEY` seems (very) expensive.

Reply:
> Hmm, this seems not correct, how is the 16 calculated? Also, `GETEY` seems (very) expensive.

Because the operation is on 256 bits of data

Reply:
it should be 24. but like I said, the GETEY is too expensive and probably wrong, just reuse the result from GETEX.

Reply:
> Running you current `GETEY` after `GETEX` will endup running `geted` 2 times for the opcode, wich might result is reading more bytes from the opcode than actually exist. You should really looks athow it's done on the ARM64 side: geted is only called at the GETEX side, not on GETEY. Also, with what program or game are you testing your changed?

It is also possible to call it only on the GETGX side, the fixedaddress address is the same, but fixedaddress +=16 is correct, offset by 128 to get the value of ymm.

Reply:
> Running you current `GETEY` after `GETEX` will endup running `geted` 2 times for the opcode, wich might result is reading more bytes from the opcode than actually exist. You should really looks athow it's done on the ARM64 side: geted is only called at the GETEX side, not on GETEY. Also, with what program or game are you testing your changed?

A test demo I wrote myself

#include <stdio.h>

int main() {
    float src[8] = {1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f};
    float dst[8] = {0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f}; 

    __asm__ volatile (
        "vmovups (%1), %%ymm0\n"  
        "vmovups %%ymm0, (%0)\n"  
        :                        
        : "r" (dst), "r" (src)  
        : "%ymm0" 
    );

    printf("Source data: %f %f %f %f %f %f %f %f\n", src[0], src[1], src[2], src[3], src[4], src[5], src[6], src[7]);
    printf("Destination data: %f %f %f %f %f %f %f %f\n", dst[0], dst[1], dst[2], dst[3], dst[4], dst[5], dst[6], dst[7]);

    return 0;
}
~

Reply:
looks like a valid sample code (did you look at the box64 dump to see what is generated code for it?).
Feel free to add a few more opcode and add this the box64 test suite if it makes sense.

Reply:
> looks like a valid sample code (did you look at the box64 dump to see what is generated code for it?). Feel free to add a few more opcode and add this the box64 test suite if it makes sense.

Ok, geted should indeed be executed once, I've made the change

Reply:
> looks like a valid sample code (did you look at the box64 dump to see what is generated code for it?). Feel free to add a few more opcode and add this the box64 test suite if it makes sense.
![Uploading 2024-10-28_17-33-11.png…]()

Reply:
> looks like a valid sample code (did you look at the box64 dump to see what is generated code for it?). Feel free to add a few more opcode and add this the box64 test suite if it makes sense.

![2024-10-28_17-33-11](https://github.com/user-attachments/assets/0a00f5d7-deb0-41ca-9757-83cf5ecd3db3)

Reply:
So, the VMOVUPS didn't have SIB or Offset involved in this sample, that's why calling 2* times geted still worked. It would have crash and burn if an offset was involved in the opcode for example. Maybe try to use only 1 single array, instead of two?

(pro tips: use `less -R dump.txt` to see the dump in all it's colorfull glory)